ledfx/utils.py

```
@@ -1799,20 +1799,164 @@ def validate_local_path(file_path: str) -> tuple[bool, str | None]:
+            )
+            return None
+
+    # Validate MIME type
+    if not validate_image_mime_type(resolved_path):
+        _LOGGER.error(f"Invalid image MIME type: {original_path}")
+        return None
+
+    # Open the image
+    try:
+        gif = Image.open(resolved_path)
+    except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)